src/hooks/useNearbyPlaces.ts

```diff
@@ -25,17 +25,25 @@ const hasSameCoordinates = (a?: LocationObject, b?: LocationObject): boolean =>
 	);
 };
 
-export const useNearbyPlaces = (): { nearbyPlaces?: PlaceWithAccesibilityData[] } => {
+export const useNearbyPlaces = (
+	keyword: string
+): { nearbyPlaces?: PlaceWithAccesibilityData[] } => {
 	const { location } = useLocation();
 	const lastLocationRef = React.useRef<LocationObject | undefined>(undefined);
+	const keywordRef = React.useRef<string | undefined>(undefined);
 	const [nearbyPlaces, setNearbyPlaces] = React.useState<PlaceWithAccesibilityData[]>();
 
 	const getNearbyPlaces = async (location: LocationObject) => {
-		if (hasSameCoordinates(location, lastLocationRef.current)) return;
+		if (hasSameCoordinates(location, lastLocationRef.current) && keyword === keywordRef.current)
+			return;
+		setNearbyPlaces(undefined);
 		const result = await axios.get<{ places: PlaceWithAccesibilityData[] }>(
-			`${SERVER_BASE_URL}/getNearbyPlaces?latitude=${location.coords.latitude}&longitude=${location.coords.longitude}&includeRatings=true`
+			`${SERVER_BASE_URL}/getNearbyPlaces?latitude=${location.coords.latitude}&longitude=${
+				location.coords.longitude
+			}&includeRatings=true${keyword && keyword.length > 0 ? `&keyword=${keyword}` : ""}`
```

Comment:
shouldn't we be using `type` instead of `keyword` here? when designing the `placeTypes` I had these in mind:

https://developers.google.com/maps/documentation/places/web-service/supported_types#table1
https://developers.google.com/maps/documentation/places/web-service/search-nearby#type

Reply:
i debated that but i went with keyword because keyword is more flexible so we could technically add our own types, but now i'm rethinking, because we're planning on adding search eventually. i'll change it